pandas/core/groupby/ops.py

```diff
@@ -543,6 +543,17 @@ def _cython_operation(
             if mask.any():
                 result = result.astype("float64")
                 result[mask] = np.nan
+        elif (
```

Comment:
Ugly! But this ensures that `Series[IntDtype].resample().sum()` has int dtype. Assuming we want the restriction "only try casting back to EA when the scalars results are instances of the EA dtypes type, then I think this is the least worst option.

I briefly tried avoiding the cast from Int64 -> float with NaN, but that wasn't feasible today. Our Cython reductions like `group_add` only handle floats IIUC.

Reply:
TODO comment suggesting this get cleaned up, pointing back to this thread?

Reply:
Going to open an issue and will point here.